src/role_nextroomer.js

```diff
@@ -308,11 +308,15 @@ roles.nextroomer.settle = function(creep) {
     methods.push(Creep.constructTask);
   }
 
+  // Fill energy after building is done
+  if (creep.room.controller.level >= 4) {
+    methods.push(Creep.transferEnergy);
+  }
+
   if (creep.room.controller.level < 8) {
+
```

Comment:
the problem is the lack of energy. try https://github.com/TooAngel/screeps/pull/98 i think it could resolve nextroom which harvest to slowly and build roads by the same.